core/utils/ethabi.go

```diff
@@ -26,21 +27,31 @@ const (
 	FormatBool = "bool"
 )
 
-// GenericEncode eth encodes values based on the provided types
-func GenericEncode(types []string, values ...interface{}) ([]byte, error) {
-	if len(values) != len(types) {
-		return nil, errors.New("must include same number of values as types")
+// ABIEncode is the equivalent of abi.encode.
```

Comment:
Ended up not strictly needing this, but I think its useful and could potential simplify the pipeline abi encoding / log parsing tasks